include/manager.hpp

```diff
@@ -40,6 +40,21 @@ class Manager
     ~Manager() = default;
 
   private:
+#ifdef IBM_SYSTEM
+    /**
+     * @brief API to set timer to detect system VPD collection status.
+     *
+     * System VPD is required before bus for VPD-Manager is claimed. Once system
+     * VPD is published, VPD for other FRUs should be collected. This API
+     * detects system VPD collection status and triggers VPD collection for rest
+     * of the FRUs.
+     *
+     * Note: Throws exception in case of any failure. Needs to be handled by the
+     * caller.
+     */
+    void setTimerForSystemVPDDetection();
```

Comment:
setTimerForSystemVPDDetection -> I think this timer is for system vpd collection(as per the implementation ), and not for system vpd detection (as name says).
Detecting system vpd collection status and Detecting System vpd are 2 different things.